lib/engine/game/g_18_india/step/home_track.rb

```diff
@@ -48,29 +48,29 @@ def process_lay_tile(action)
           end
 
           def process_place_token(action)
-            LOGGER.debug 'HomeTrack > process_place_token'
+            LOGGER.debug { 'HomeTrack > process_place_token' }
             super
             tile = action.city.tile
+            LOGGER.debug { "HomeTrack > tile: #{tile.inspect}" }
             replace_oo_reservations(tile) unless tile.reservations.empty? # move hex reservation
           end
 
           # Base code doesn't handle one token and one reservation on a OO tile
           # Moves a reservation from hex to untoken city
           def replace_oo_reservations(tile)
-            return if tile.reservations.empty?
+            return unless tile.label == 'OO'
 
             cities = tile.cities
             reservations = tile.reservations.dup
-            LOGGER.debug "replace_oo_reservations > reservations: #{reservations}"
-            tile.reservations.each do |reservation|
-              cities.each do |city|
-                if !city.tokened? || cities.count == 1
-                  city.add_reservation!(reservation)
-                  exit
-                end
-              end
+            LOGGER.debug { "replace_oo_reservations > reservations: #{reservations}" }
+            cities.each do |city|
+              next if city.tokened?
+
+              corp = reservations.pop
+              tile.remove_reservation!(corp)
+              city.add_reservation!(corp)
+              LOGGER.debug { "replace_oo_reservations > city reservation: #{city.reservations}" }
             end
```

Comment:
If I've understood this right, this will only get called when a yellow OO tile is being placed, and there will be one tile reservation, and two cities – one with a token and one empty. The reservation needs to be moved from the tile to the open city. Is that correct?

What was wrong with the older version of the code? The `cities.count == 1` test?

I don't see the need for a loop here. I guess that you're trying to generalise the code here, but this would fail if it got called when there were more open cities than tile reservations. You either need to add some more checks, or simplify it based on what we know.

```
corp = tile.reservations.first
city = cities.reject(&:tokened?).first
city.add_reservation!(corp)
tile.reservations.clear
```

Reply:
Modified method to remove loop as suggested. Thanks!